### PR TITLE
docs: Add working zsh completion instructions

### LINKED ISSUE #10534: zsh autocompletion doesn't work
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

zsh autocompletion doesn't work, with the instructions given in https://github.com/argoproj/argo-cd/blob/master/docs/user-guide/commands/argocd_completion.md

**To Reproduce**

- add the output of argocd completion zsh to a file in $fpath
- try to do tab completion
- doesn't work

**Expected behavior**

for completion to work in zsh

**Version**

```shell
argocd: v2.4.11+3d9e9f2
  BuildDate: 2022-08-22T09:35:38Z
  GitCommit: 3d9e9f2f95b7801b90377ecfc4073e5f0f07205b
  GitTreeState: clean
  GoVersion: go1.18.5
  Compiler: gc
  Platform: linux/amd64
```

### PR DESCRIPTION
**Pull request by dimisjim**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Closes https://github.com/argoproj/argo-cd/issues/10534

### CODE REVIEW

**Review by crenshaw-dev — COMMENTED** [review]
@dimisjim thanks! Can you run `make clidocsgen` and push again?

**Comment by dimisjim** [comment]
@crenshaw-dev don't

File: cmd/argocd/commands/completion.go [hunk 1]
```diff
@@ -203,8 +203,9 @@ To access completions in your current shell, run
 $ source <(argocd completion bash)
 Alternatively, write it to a file and source in .bash_profile
 
-For zsh, output to a file in a directory referenced by the $fpath shell
-variable.
+For zsh, add the following to your ~/.zshrc file:
+source <(argocd completion zsh)
+compdef _argocd argocd
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/completion.go:208]: @dimisjim I might be doing something silly, but I'm getting this error:

```
% zsh
/Users/mcrenshaw/.zshrc:3: command not found: compdef
```

Comment (dimisjim) [reply cmd/argocd/commands/completion.go:208]: are you running oh my zsh?
maybe it's a bit different on mac? (I am running it on linux)

have you tried adding to your ~/.zshrc file the following:

```
autoload -Uz compinit
compinit
```

as suggested [here](https://unix.stackexchange.com/a/477909/306647)?

Comment (crenshaw-dev) [reply cmd/argocd/commands/completion.go:208]: That works! So the complete solution for my Mac is

```
autoload -Uz compinit
compinit
source <(argocd completion zsh)
compdef _argocd argocd
```

Do you think it's worth adding that to the docs, or think it's enough a niche issue to justify leaving it out? Folks can always search the error message.

Comment (dimisjim) [reply cmd/argocd/commands/completion.go:208]: I think it would be nice to add it to the docs.

Comment (crenshaw-dev) [reply cmd/argocd/commands/completion.go:208]: Sounds good! If you update the PR, I'll happily merge.  :-)

Comment (dimisjim) [reply cmd/argocd/commands/completion.go:208]: sure, would this suffice? https://github.com/argoproj/argo-cd/pull/10535/commits/b87e6e4e7433545064f1b19a56819058c66d08bb

Comment (crenshaw-dev) [reply cmd/argocd/commands/completion.go:208]: Yep, that works! 

argocd_completion is generated from completion.go. You'll have to add the text in completion.go and then run `make clidocsgen-local`.

Comment (dimisjim) [reply cmd/argocd/commands/completion.go:208]: Sure, ran `make clidocsgen`, made sure commits are signed, and I guess it should be good now.

**Comment by dimisjim** [comment]
hey @crenshaw-dev Does it look good?

**Review by blakepettersson — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]
Thanks @dimisjim!

**Pull request merged**